### PR TITLE
[firebase_auth] Adding support for FirebaseUser.unlink(providerId)

### PR DESCRIPTION
**Pull request by ko2ic**
* Unlink an auth provider from a user account

### CODE REVIEW

**Comment by ko2ic** [comment]
The difference from #775 is whether the method is divided for each Provider or whether to supply ProviderId as argument. 
(Other differences are here, PR of one modification.)
If it is better to separate the method for each provider, we will fix it.

**Comment by kroikie** [comment]
@ko2ic Thanks for this PR, please rebase onto master so that conflicts are resolved. Also could you limit your changes to the unlink feature? If you want to do other edits please submit those in another PR.

**Comment by ko2ic** [comment]
> Also could you limit your changes to the unlink feature?

Of course. I will do so.

File: packages/firebase_auth/ios/Classes/FirebaseAuthPlugin.m [hunk 1]
```diff
@@ -227,6 +237,10 @@ - (void)handleMethodCall:(FlutterMethodCall *)call result:(FlutterResult)result
             completion:^(NSError *error) {
               [self sendResult:result forUser:nil error:error];
             }];
+    [[FIRAuth auth].currentUser updatePassword:password
```
Comment (kroikie) [inline packages/firebase_auth/ios/Classes/FirebaseAuthPlugin.m]: Could we handle this in another PR. Removing this and only handling unlink in this PR. Thanks.

Comment (ko2ic) [reply packages/firebase_auth/ios/Classes/FirebaseAuthPlugin.m]: I fixed it.
Could you review again?

Comment (ko2ic) [reply packages/firebase_auth/ios/Classes/FirebaseAuthPlugin.m]: You may not know the change point with force push.
The changed point is two.
* I fixed it to handle only A.
* [FIRAuth auth] -> [self getAuth:call.arguments] in ```unlink```.

Comment (manhnhhdvn) [reply packages/firebase_auth/ios/Classes/FirebaseAuthPlugin.m]: @kroikie : I want to use this unlink function. Could you review this pull request as soon as possible?

**Comment by ko2ic** [comment]
@kroikie Conflict resolved.
Merge is now available.

**Comment by ko2ic** [comment]
@kroikie I did it.

**Pull request closed without merging**